examples/run_smc_on_freesolv.py

```diff
@@ -179,16 +196,23 @@ def main():
     num_gpus = cmd_args.n_gpus or 1
     client = CUDAPoolClient(max_workers=num_gpus)
     client.verify()
+    file_client = FileClient(base=Path(cmd_args.result_path))
     print(f"using {num_gpus} gpus with {cmd_args.n_cpus or 'default'} cpus per gpu")
 
     # Batch mols
+    ff = get_ff(cmd_args.ff)
     batch_mols = batch_list(mols, num_gpus)
     futures = []
     for mol_subset in batch_mols:
-        futures.append(client.submit(run_on_mols, mol_subset, cmd_args))
+        futures.append(client.submit(run_on_mols, file_client, mol_subset, ff, cmd_args))
 
     # Wait for jobs to complete
-    _ = [fut.result() for fut in futures]
+    batched_results = [fut.result() for fut in futures]
+    results = [i for j in batched_results for i in j]
```

Comment:
nit: naming might improve readability
```suggestion
    results = [result for batch in batched_results for result in batch]
```